### DISCUSSION TITLE
Customizing Masks in Charm for Head Mesh Generation

### DISCUSSION THREAD

**Discussion by savvy73** [simnibs/simnibs#286] [7.Creating head models]
I have different masks for various parts of the head, such as hair, air, white matter (WM), gray matter (GM), etc. Can I replace the default masks in Charm with my own masks? I want to run the Charm command to generate the mesh after substituting these masks. Will Charm still work effectively with custom masks? your answers are very much appriciated

**Comment by oulap** [comment]
Please have a look at these examples:

https://simnibs.github.io/simnibs/build/html/documentation/command_line/add_tissues_to_upsampled.html

https://simnibs.github.io/simnibs/build/html/tutorial/advanced/fix_headmodel.html

**Reply by savvy73** [reply]
Hi, I already tried that. I just have masks that have to be added together to get a head model. I was trying to do one of the following:
1. I am trying to replace the masks from the Ernie dataset to get the file tissue_labeling_unsampled  which can be later used to generate mesh or 
2. I want to add those masks and assign conductivity to create a file like tissue_labeling_unsampled which can later be used to mesh

**Reply by oulap** [reply]
If you add the masks together with different labels to get a segmentation you can then just use this:

https://simnibs.github.io/simnibs/build/html/documentation/command_line/meshmesh.html#meshmesh

command to create a mesh (from any label image). The conductivity assignment is done when you setup the simulation, see example here:

https://simnibs.github.io/simnibs/build/html/tutorial/advanced/custom_meshing.html

**Reply by savvy73** [reply]
Thank you for your response. What method would you suggest to add masks together with different labels? 
Looking forward to your reply

**Reply by oulap** [reply]
You can use python or matlab or whatever other software you prefer to combine the masks. If the masks do *not* overlap you can just multiply each with a label you choose and then combine those to a single file. If the masks *do* overlap you need to choose the order which you combine the masks as the "latest" mask will overwrite the shared voxels.
